corehq/apps/app_manager/xform.py

```diff
@@ -495,32 +496,45 @@ def update_block(self):
         self.elem.append(update_block)
         return update_block
 
-    def add_case_updates(self, updates, make_relative=False):
+    def add_case_updates(self, updates, make_relative=False, domain=None):
+        from corehq.apps.app_manager.models import ConditionalCaseUpdate
         update_block = self.update_block
         if not updates:
             return
 
         update_mapping = {}
         attachments = {}
         for key, value in updates.items():
+            value_str = getattr(value, "question_path", value)
             if key == 'name':
                 key = 'case_name'
-            if self.is_attachment(value):
-                attachments[key] = value
+            if self.is_attachment(value_str):
+                attachments[key] = value_str
             else:
                 update_mapping[key] = value
 
         for key, q_path in sorted(update_mapping.items()):
+            resolved_path = self.xform.resolve_path(q_path)
+            edit_mode_path = ''
+            if (SAVE_ONLY_EDITED_FORM_FIELDS.enabled(domain, 'domain')
```

Comment:
Because the FF check occurs in `add_case_updates` in `XFormCaseBlock` we'd still need to pass `conditional_case_updates=conditional_case_updates` in each call of `add_case_updates`. If you meant add to the constructor of `XFormCaseBlock`, that would mean we would need to add more FF checks, one for each time we create an `XFormCaseBlock` and call `add_case_updates`. Would you still consider that cleaner?

Reply:
@snopoke tagging you for visibility